spring-boot-tests/spring-boot-smoke-tests/spring-boot-smoke-test-pulsar/src/test/java/smoketest/pulsar/SamplePulsarApplicationTests.java

```diff
@@ -50,17 +54,40 @@ static void pulsarProperties(DynamicPropertyRegistry registry) {
 		registry.add("spring.pulsar.admin.service-url", PULSAR_CONTAINER::getHttpServiceUrl);
 	}
 
-	@Test
-	void appProducesAndConsumesSampleMessages(@Autowired SampleMessageConsumer consumer) {
-		Awaitility.await()
-			.atMost(Duration.ofMinutes(3))
-			.with()
-			.pollInterval(Duration.ofMillis(500))
-			.untilAsserted(() -> hasExpectedIds(consumer));
+	@Nested
+	@SpringBootTest
+	@ActiveProfiles("smoketest.pulsar.imperative")
+	class ImperativeApp {
+
+		@Test
+		void appProducesAndConsumesMessages(CapturedOutput output) {
+			List<String> expectedOutput = new ArrayList<>();
+			IntStream.range(0, 10).forEachOrdered((i) -> {
+				expectedOutput.add("++++++PRODUCE IMPERATIVE:(" + i + ")------");
+				expectedOutput.add("++++++CONSUME IMPERATIVE:(" + i + ")------");
+			});
+			Awaitility.waitAtMost(Duration.ofSeconds(30))
```

Comment:
Using captured output simplifies by removing the need for the message consumer component and the list of messages etc..